cmd/warpforge/trace.go

```diff
@@ -0,0 +1,99 @@
+package main
+
+import (
+	"context"
+	"io"
+	"os"
+
+	"github.com/warpfork/warpforge/wfapi"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/exporters/stdout/stdouttrace"
+	"go.opentelemetry.io/otel/sdk/resource"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	semconv "go.opentelemetry.io/otel/semconv/v1.12.0"
+)
+
+const TRACER_NAME = "main"
+
+// configTracer sets the default tracing configuration
+// The caller must call Shutdown on the provider
+func configTracer(filename string) (*sdktrace.TracerProvider, error) {
+	res, err := resource.Merge(
+		resource.Default(),
+		resource.NewWithAttributes(
+			semconv.SchemaURL,
+			semconv.ServiceNameKey.String(MODULE),
+			semconv.ServiceVersionKey.String(VERSION),
+		),
+	)
+	if err != nil {
+		return nil, err
+	}
+	exp, err := newFileSpanExporter(filename)
+	if err != nil {
+		return nil, err
+	}
+	if exp == nil {
+		return nil, nil
+	}
+	tp := sdktrace.NewTracerProvider(
+		sdktrace.WithSampler(sdktrace.AlwaysSample()),
+		sdktrace.WithBatcher(exp),
+		sdktrace.WithResource(res),
+	)
+	otel.SetTracerProvider(tp)
+	return tp, nil
+}
+
+func traceShutdown(ctx context.Context, tp *sdktrace.TracerProvider) error {
+	if tp == nil {
+		return nil
+	}
+	return tp.Shutdown(ctx)
+}
+
+// fileSpanExporter calls Close() during Shutdown, simplifying the
+// implementation for file handling
+type fileSpanExporter struct {
+	sdktrace.SpanExporter
+	io.Closer
+}
+
+// Shutdown handles cleaning up the span exporter
+// Errors:
+//
+//     - warpforge-error-unknown -- when an error occurs during tracing shutdown
+func (e *fileSpanExporter) Shutdown(ctx context.Context) error {
```

Comment:
Such aggressive linting on error handling...!

I think a fairly catch-all error code for this is fine, because if something goes wrong with the telemetry stuff, it's almost certainly a "operator intervention required, after they read the fulltext message" scenario.  `warpforge-error-internal` or something.